From 1e99cb806fb7d6a341bf8f7c238aa7261af7dbd0 Mon Sep 17 00:00:00 2001
From: Sebastian Ott <sebott@redhat.com>
Date: Wed, 29 Oct 2025 15:53:41 +0100
Subject: [PATCH 06/10] arm: create new rhel 10.2 specific virt machine type

RH-Author: Sebastian Ott <sebott@redhat.com>
RH-MergeRequest: 416: x86, arm: create new rhel 9.8, 10.2 specific machine types
RH-Jira: RHEL-105826 RHEL-105828
RH-Acked-by: Eric Auger <eric.auger@redhat.com>
RH-Acked-by: Gavin Shan <gshan@redhat.com>
RH-Acked-by: Cornelia Huck <cohuck@redhat.com>
RH-Commit: [1/4] 829eed93aefddcc4684ddacff3cd6ff3f991f442 (seott1/cos-qemu-kvm)

Signed-off-by: Sebastian Ott <sebott@redhat.com>
---
 hw/arm/virt.c | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/hw/arm/virt.c b/hw/arm/virt.c
index d37d1bb3cf..e68979d2c5 100644
--- a/hw/arm/virt.c
+++ b/hw/arm/virt.c
@@ -3688,16 +3688,23 @@ static void virt_machine_4_1_options(MachineClass *mc)
 DEFINE_VIRT_MACHINE(4, 1)
 #endif /* disabled for RHEL */
 
+static void virt_rhel_machine_10_2_0_options(MachineClass *mc)
+{
+}
+DEFINE_VIRT_MACHINE_AS_LATEST(10, 2, 0)
+
 static void virt_rhel_machine_10_0_0_options(MachineClass *mc)
 {
     VirtMachineClass *vmc = VIRT_MACHINE_CLASS(OBJECT_CLASS(mc));
 
+    virt_rhel_machine_10_2_0_options(mc);
+
     /* QEMU 9.1 and earlier have only a stage-1 SMMU, not a nested s1+2 one */
     vmc->no_nested_smmu = true;
     compat_props_add(mc->compat_props, hw_compat_rhel_10_2, hw_compat_rhel_10_2_len);
     compat_props_add(mc->compat_props, hw_compat_rhel_10_1, hw_compat_rhel_10_1_len);
 }
-DEFINE_VIRT_MACHINE_AS_LATEST(10, 0, 0)
+DEFINE_VIRT_MACHINE(10, 0, 0)
 
 static void virt_rhel_machine_9_6_0_options(MachineClass *mc)
 {
-- 
2.47.3

